### PR TITLE
Evaka hotfix sort attendances

### PR DESCRIPTION
**Pull request by massproduction**
#### Summary

Sort child attendances children by first name.

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt [hunk 1]
```diff
@@ -341,7 +341,7 @@ private fun Database.Read.getAttendancesResponse(unitId: UUID): AttendanceRespon
         )
     }
 
-    return AttendanceResponse(unitInfo, children)
+    return AttendanceResponse(unitInfo, children.sortedBy { it.firstName })
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt:344]: I think others have encouraged sorting in frontend. You can use lodash `_,sortyBy(children, child => child.name)`

Comment (massproduction) [reply service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt]: Moved the sorting to the frontend.

File: frontend/packages/employee-frontend/src/api/attendances.ts [hunk 2]
```diff
@@ -201,13 +201,26 @@ export async function postDeparture(
...
+}
+
 function deserializeAttendanceResponse(
   data: JsonOf<AttendanceResponse>
 ): AttendanceResponse {
   {
     return {
       unit: data.unit,
-      children: data.children.map((attendanceChild) => {
+      children: data.children.sort(compareFirstNames).map((attendanceChild) => {
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/api/attendances.ts:223]: Now two children with same first name will have undefined order, wouldn't it make more sense to then sort by last name next? Also a custom comparator for such a simple case is quite verbose compared to using lodash.

File: frontend/packages/employee-frontend/src/api/attendances.ts [hunk 3]
```diff
@@ -201,13 +201,26 @@ export async function postDeparture(
     .catch((e) => Failure.fromError(e))
 }
 
+function compareFirstNames(
+  a: JsonOfObject<AttendanceChild>,
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/api/attendances.ts:205]: do we need `JsonOfObject` here?

Comment (massproduction) [reply frontend/packages/employee-frontend/src/api/attendances.ts:205]: Yeah, it was giving a warning without it.

**Review by Joosakur — APPROVED** [review]
only minor comments 👍

File: frontend/packages/employee-frontend/src/api/attendances.ts (hunk 3 above)
Comment (Joosakur) [reply frontend/packages/employee-frontend/src/api/attendances.ts:205]: huh, weird

File: frontend/packages/employee-frontend/src/api/attendances.ts [hunk 4]
```diff
@@ -220,20 +221,23 @@ function deserializeAttendanceResponse(
...
-                departed: attendanceChild.attendance.departed
-                  ? new Date(attendanceChild.attendance.departed)
-                  : null
-              }
-            : null
-        }
-      })
+      children: data.children
+        .sort((a, b) => compareByProperty(a, b, 'lastName'))
+        .sort((a, b) => compareByProperty(a, b, 'firstName'))
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/api/attendances.ts:226]: fwiw alternative option would be `_.sortBy(data.children, ['firstName', 'lastName']).map( ...`

**Pull request merged**